qabelbox/src/main/java/de/qabel/qabelbox/box/interactor/BoxProviderUseCase.kt

```diff
@@ -0,0 +1,66 @@
+package de.qabel.qabelbox.box.interactor
+
+import de.qabel.qabelbox.box.dto.*
+import de.qabel.qabelbox.box.provider.DocumentId
+import rx.Observable
+import rx.lang.kotlin.toSingletonObservable
+import java.io.FileNotFoundException
+import javax.inject.Inject
+
+class BoxProviderUseCase @Inject constructor(private val volumeManager: VolumeManager):
+        ProviderUseCase {
+
+    override fun query(documentId: DocumentId): Observable<BrowserEntry> {
+        return browserByDocumentId(documentId).query(documentId.path)
+    }
+
+    override fun availableRoots(): List<VolumeRoot> = volumeManager.roots
+
+    override fun queryChildDocuments(documentId: DocumentId): Observable<List<ProviderEntry>> {
+        when (documentId.path) {
+            is BoxPath.File -> return emptyList<ProviderEntry>().toSingletonObservable()
+            is BoxPath.FolderLike -> {
+                return browserByDocumentId(documentId).list(documentId.path).map { entries ->
+                    entries.map {
+                        val path = when (it) {
+                            is BrowserEntry.File -> documentId.path * it.name
+                            is BrowserEntry.Folder -> documentId.path / it.name
+                        }
+                        ProviderEntry(documentId.copy(path = path), it)
+                    }
+                }
+            }
```

Comment:
extracting this anonymous method to something named would increase redability here ... and that's neccessary